pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BacklogQuotaManager.java

```diff
@@ -268,4 +271,29 @@ private void disconnectProducers(PersistentTopic persistentTopic) {
 
         });
     }
+
+    private void advanceSlowestMessageDeduplicationCursor(PersistentTopic persistentTopic,
+                                                          BacklogQuotaType backlogQuotaType) {
+
+        if (backlogQuotaType != destination_storage) {
+            return;
+        }
+
+        MessageDeduplication dedup = persistentTopic.getMessageDeduplication();
+        if (dedup == null) {
+            return;
+        }
+
+        ManagedLedgerImpl mLedger = (ManagedLedgerImpl) persistentTopic.getManagedLedger();
+        ManagedCursor slowestConsumer = mLedger.getSlowestConsumer();
+        if (slowestConsumer == null) {
+            return;
+        }
+
+        if (!PersistentTopic.isDedupCursorName(slowestConsumer.getName())) {
+            return;
+        }
+
+        dedup.takeSnapshot();
```

Comment:
We also have an interval check at [here](https://github.com/apache/pulsar/pull/20951/files#diff-32e651dda4c98296475fd4e3cd322c76d6da3618c1013eda21c6b9033dec1bc1R502-R505)

As I understand, you want to take a snapshot forcefully, no matter the value of the interval. Otherwise, the dedup cursor still can't move forward.

Reply:
And it's better to have test to cover this case.

Reply:
The system cursor should never cause the producer disconnection because the backlog quota is a user-facing policy. Users should not care about the system cursors when configuring the backlog quota policy. So even if users had provided a meaningless time-based snapshot duplication policy. The producer shouldn't be blocked.

Of course, honoring the interval check and the latest changes looks good to me. Instead of taking snapshots too frequently, we let the backlogs run over the quota. It makes more sense because we should choose the policy with minimum impact on the production environment.